packages/node-core/src/indexer/project.service.ts

```diff
@@ -198,7 +200,7 @@ export abstract class BaseProjectService<
 
     this.eventEmitter.emit(IndexerEvent.NetworkMetadata, this.apiService.networkMeta);
 
-    const keys = [
+    const keys: (keyof MetadataKeys)[] = [
```

Comment:
I think `ensureMetadata` should probably live in cacheMetadata and be called from here instead. Its out of scope of this PR